### ISSUE TITLE
[deckhouse] deckhouse init-container image doesn't change when changing d8 version FE → CE

### ISSUE THREAD

**Issue by meatuses** [deckhouse/deckhouse#4811]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v.1.45.11

### Expected Behavior

On FE d8 cluster, after step 2 of this doc https://deckhouse.io/documentation/v1/deckhouse-faq.html#how-to-switch-deckhouse-ee-to-ce 
```
bash -c "$(curl -Ls https://raw.githubusercontent.com/deckhouse/deckhouse/main/tools/change-registry.sh)" -- --registry-url https://registry.deckhouse.io/deckhouse/ce && \
kubectl -n d8-system set image deployment/deckhouse deckhouse=$(kubectl -n d8-system get deployment deckhouse -o jsonpath='{.spec.template.spec.containers[?(@.name=="deckhouse")].image}' | awk -F: '{print "registry.deckhouse.io/deckhouse/ce:" $2}')
```
Deckhouse pod should be recreated and start correctly using CE images

### Actual Behavior

While following this doc https://deckhouse.io/documentation/v1/deckhouse-faq.html#how-to-switch-deckhouse-ee-to-ce after the step 2, deckhouse container can't start.
Pod events:
```
kubectl describe pod -n d8-system  deckhouse-78dd5b7d76-pfwc2
Events:  
Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  8m18s                  default-scheduler  Successfully assigned d8-system/deckhouse-78dd5b7d76-pfwc2 to downgrade
  Normal   Pulling    6m39s (x4 over 8m17s)  kubelet            Pulling image "registry.deckhouse.io/deckhouse/fe@sha256:79ed551f4d0ec60799a9bd67f35441df6d86443515dd8337284fb68d97a01b3d"
  Warning  Failed     6m39s (x4 over 8m17s)  kubelet            Failed to pull image "registry.deckhouse.io/deckhouse/fe@sha256:79ed551f4d0ec60799a9bd67f35441df6d86443515dd8337284fb68d97a01b3d": rpc error: code = Unknown desc = failed to pull and unpack image "registry.deckhouse.io/deckhouse/fe@sha256:79ed551f4d0ec60799a9bd67f35441df6d86443515dd8337284fb68d97a01b3d": failed to resolve reference "registry.deckhouse.io/deckhouse/fe@sha256:79ed551f4d0ec60799a9bd67f35441df6d86443515dd8337284fb68d97a01b3d": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
  Warning  Failed     6m39s (x4 over 8m17s)  kubelet            Error: ErrImagePull
  Warning  Failed     6m27s (x6 over 8m16s)  kubelet            Error: ImagePullBackOff
  Normal   BackOff    3m8s (x20 over 8m16s)  kubelet            Back-off pulling image "registry.deckhouse.io/deckhouse/fe@sha256:79ed551f4d0ec60799a9bd67f35441df6d86443515dd8337284fb68d97a01b3d"
```
init-external-modules container keeps its' FE tag:
```
Init Containers:
  init-external-modules:
    Container ID:
    Image:         registry.deckhouse.io/deckhouse/fe@sha256:79ed551f4d0ec60799a9bd67f35441df6d86443515dd8337284fb68d97a01b3d
```

### Steps To Reproduce

1. Fully set up d8 cluster v1.45.11 FE (bare metal)
2. Run this command on master node: 
```
bash -c "$(curl -Ls https://raw.githubusercontent.com/deckhouse/deckhouse/main/tools/change-registry.sh)" -- --registry-url https://registry.deckhouse.io/deckhouse/ce && \
kubectl -n d8-system set image deployment/deckhouse deckhouse=$(kubectl -n d8-system get deployment deckhouse -o jsonpath='{.spec.template.spec.containers[?(@.name=="deckhouse")].image}' | awk -F: '{print "registry.deckhouse.io/deckhouse/ce:" $2}')
```
3. Deckhouse pod can't start
```
# kubectl -n d8-system get  po -l app=deckhouse
NAME                         READY   STATUS                  RESTARTS   AGE
deckhouse-78dd5b7d76-pfwc2   0/1     Init:ImagePullBackOff   0          18m
```
4. Try command suggested in step 4 of linked doc, but outcome doesn't change:
`kubectl -n d8-system delete po -l app=deckhouse`

**Comment by name212** [comment]
duplicate of https://github.com/deckhouse/deckhouse/issues/4326?
